crawl-ref/source/items.cc

```diff
@@ -1841,6 +1841,8 @@ static void _get_book(const item_def& it, bool quiet)
                 spellnames.push_back(spell_title(st));
             else
                 you.hidden_spells.set(st, true);
+            if (Options.auto_hide_spells && allow_auto_hide)
+                you.hidden_spells.set(st, true);
```

Comment:
I don't love the duplication here, but I don't see a clean way to remove it. Maybe:

    if (memorise(...))
      ...;
    if (hide || !memorise(...))
      hide();

Duplicates the can_memorise check instead, which seems a bit better to me but I wouldn't mind if you disagreed.

Reply:
```
if (memorize)
{
  spell_names.push_back();
  if (hide)
    hide();
}
else
  hide();
```

Removes the duplication in execution but is a bit less easy to read. Thoughts?